ROADMAP.md

```diff
@@ -0,0 +1,92 @@
+# io.js Roadmap
+
+***This is a living document, it describes the policy and priorities as they exist today but can evolve over time.***
+
+## Stability Policy
+
+The single more important consideration in every code change is the impact it will have, positive or negative, on the ecosystem (modules and applications).
+
+io.js does not remove stdlib JS API.
+
+Shipping with current and well supported dependencies is the best way to ensure long term stability of the platform. When those dependencies are no longer maintained io.js will take on their continued maintenance as part of our [Long Term Support policy](#long-term-support).
+
+io.js will continue to adopt new V8 releases.
+* When V8 ships a breaking change to their C++ API that can be handled by [`nan`](https://github.com/rvagg/nan)
+the *minor* version of io.js will be increased.
+* When V8 ships a breaking change to their C++ API that can NOT be handled by [`nan`](https://github.com/rvagg/nan)
+the *major* version of io.js will be increased.
+* When new features in the JavaScript language are introduced by V8 the
+*minor* version number will be increased. TC39 has stated clearly that no
+backwards incompatible changes will be made to the language so it is
+appropriate to increase the minor rather than major.
+
+No new API will be added in *patch* releases.
+
+Any API addition will cause an increase in the *minor* version.
+
+### Long Term Support
+
+io.js supports old versions for as long as community members are fixing bugs in them.
+
+As long as there is a community back porting bug fixes we will push patch releases for those versions of io.js.
+
+When old versions of dependencies like V8 are no longer supported by their project io.js will take on the responsibility of maintenance to ensure continued long term support in io.js patch releases.
+
+## Channels
+
+* Release - Stable production ready builds. Unique version numbers following semver.
+* Canary - Nightly builds on next-generation V8 + changes landing to io.js. No version designation.
+* NG - "Next Generation." No version designation.
+
+## NG (Next Generation)
+
+In order for io.js to stay competitive we need to work on the next generation of the platform that can more accurately integrate and reflect the advancements in the language and the ecosystem.
+
+While this constitutes a great leap forward for the platform we will be making this leap without breaking backwards compatibility with the existing ecosystem of modules.
+
+NG will use ES6 modules and will be implementing a new platform and standard library available only to modules using this native new style. Modules written prior to NG using the old CommonJS module syntax will continue to operate against the old API. This is what will allow us to make improvements to the platform without breaking compatibility and still letting future NG based applications benefit from all the modules built today.
```

Comment:
> The flags enable "next generation" features in the existing platform – for instance, returning promises from async APIs that are called sans callbacks, exposing the new module syntax, or experimental changes to streams.

Extending the current API should be considered for regular releases and not block on theoretical future tech. I agree with the statement you've just made, I just don't see it as being under the same umbrella.

But, the reason I doubt we'll see something like that land is because it's such a drastic change for a fairly small gain in terms of the overall platform being friendly to new patterns. Doesn't mean we shouldn't try it out behind a flag though, especially since it's a good way to flush out issues we know we have with native promises (like we can't do AsyncWrap yet).